pytest_timeout.py

```diff
@@ -134,12 +134,34 @@ def pytest_enter_pdb():
     SUPPRESS_TIMEOUT = True
 
 
+def is_debugging():
+    """
+    Detects if a debugging session is in progress by checking if either of
+     the following conditions is true:
+        1. examining the stack frames to see if pydevd.py —
+     a debugging framework used by VSCode, PyCharm, and others — is present.
+        2. Check is SUPPRESS_TIMEOUT is set to True
+    """
+    global SUPPRESS_TIMEOUT
+    known_debugging_modules = {"pydevd"}
+    if SUPPRESS_TIMEOUT:
+        return True
+    else:
+        trace_func = sys.gettrace()
+        if trace_func:
+            for known_debugging_module in known_debugging_modules:
+                if known_debugging_module in trace_func.__module__:
```

Comment:
Perhaps `for name in know_debugging_modules` is nicer to read.  Having this long a variable name which is only one character different from the constant is not easy to read to me.

Reply:
Fair enough.